### PR TITLE
Bump @mui/monorepo

### PR DESCRIPTION
**Pull request by alexfauquette**
Mostly to get the docs update https://github.com/mui/material-ui/pull/38183

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1786,8 +1786,8 @@
     react-transition-group "^4.4.5"
 
 "@mui/monorepo@https://github.com/mui/material-ui.git#master":
-  version "5.14.2"
-  resolved "https://github.com/mui/material-ui.git#5bac8b3c66ae144c35609129e8391c1653bf0aae"
+  version "5.14.3"
+  resolved "https://github.com/mui/material-ui.git#1d1e96d1bc2542cdfd1cc9fde415fc11d682e966"
```
Comment (cherniavskii) [inline yarn.lock:1790]: Should we remove the icons?
https://github.com/mui/mui-x/blob/b27df74db171a2af42aa087a61634e1ca57b0a01/docs/data/pages.ts#L21

Screenshot 2023-08-02 at 14 16 31

Comment (alexfauquette) [reply yarn.lock:1790]: 🙈

File: yarn.lock [hunk 2]
```diff
@@ -1786,8 +1786,8 @@
     react-transition-group "^4.4.5"
 
 "@mui/monorepo@https://github.com/mui/material-ui.git#master":
-  version "5.14.2"
-  resolved "https://github.com/mui/material-ui.git#5bac8b3c66ae144c35609129e8391c1653bf0aae"
+  version "5.14.3"
```
Comment (cherniavskii) [inline yarn.lock:1789]: There's still no link to the `sx` page in https://deploy-preview-9890--material-ui-x.netlify.app/x/api/data-grid/data-grid/#DataGrid-prop-sx
Am I missing something? 😅

Comment (alexfauquette) [reply yarn.lock:1789]: fixed

the `classes` seems to have no effect, I will have to check what was the initial text added

Comment (alexfauquette) [reply yarn.lock:1789]: It's because the name is `cssApi` not classes

**Review by cherniavskii — APPROVED** [review]

**Pull request merged**